packages/kbn-text-based-editor/src/ecs_metadata_helper.test.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { getColumnsWithMetadata } from './ecs_metadata_helper';
+import type { DatatableColumnType } from '@kbn/expressions-plugin/common';
+import type { FieldsMetadataPublicStart } from '@kbn/fields-metadata-plugin/public';
+
+describe('getColumnsWithMetadata', () => {
+  it('should return original columns if fieldsMetadata is not provided', async () => {
+    const columns = [
+      { name: 'ecs.version', type: 'string' as DatatableColumnType },
+      { name: 'field1', type: 'string' as DatatableColumnType },
+      { name: 'field2', type: 'number' as DatatableColumnType },
+    ];
+
+    const result = await getColumnsWithMetadata(columns);
+    expect(result).toEqual(columns);
+  });
+
+  it('should return columns with metadata if ECS version field is present', async () => {
```

Comment:
Are we sure we want to block the descriptions based on the version field being present? I think ECS is more mix-and-match — there's no enforcement of any particular schema. 

You could have a data stream without the version field that contains many ECS fields mixed with user-defined fields. Would it not be useful to show the descriptions for matching ECS fields in this case?

cc @stratoula

Reply:
I forgot to update the test title for this but we removed the check for ecs_version. So it would call metadata service for all fields/datasets and show the description if name and type matches.

Reply:
@qn895 I'm glad to hear this. In that case, it may make sense to add a test case here that does not include an `ecs.version` field.

Reply:
@drewdaemon  I sneaked the test case into another PR here [`b53d2ed` (#188338)](https://github.com/elastic/kibana/pull/188338/commits/b53d2edbadf14552f4aff1a06ea32ea98a3c93ec)